packages/wallets/src/evm/implementations/embedded-wallet/lib/core/embedded-wallet.ts

```diff
@@ -0,0 +1,165 @@
+import type { Chain } from "@paperxyz/sdk-common-utilities";
+import { ChainToPublicRpc } from "@paperxyz/sdk-common-utilities";
+import type {
+  ClientIdWithQuerierAndChainType,
+  GetUserWalletStatusFnReturnType,
+  GetUserWalletStatusRpcReturnType,
+  SetUpWalletRpcReturnType,
+  WalletAddressObjectType,
+} from "../../interfaces/embedded-wallets/embedded-wallets";
+import { UserWalletStatus } from "../../interfaces/embedded-wallets/embedded-wallets";
+
+import { getDefaultProvider } from "ethers";
+import { LocalStorage } from "../../utils/Storage/LocalStorage";
+import type { EmbeddedWalletIframeCommunicator } from "../../utils/iFrameCommunication/EmbeddedWalletIframeCommunicator";
+import { EthersSigner } from "./signer";
+
+export type WalletManagementTypes = {
+  createWallet: void;
+  setUpNewDevice: void;
+  getUserStatus: void;
+};
+export type WalletManagementUiTypes = {
+  createWalletUi: void;
+  setUpNewDeviceUi: void;
+};
+
+export type EmbeddedWalletInternalHelperType = { showUi: boolean };
+
+export class EmbeddedWallet {
```

Comment:
@ElasticBottle  In order to make this class usable with our SDKs (React, etc) it needs to inherit from `AbstractClientWallet`. Also, note that you need a Connector for the wallet (which at this point is just re-org some code). You can look at how I did it for React Native: https://github.com/thirdweb-dev/js/pull/1561/files#diff-3b83ba028981584328b54b1c8c11a29b2101f158d827221d3127ee3ad10c2083R12